OpenRA.Mods.Common/Traits/PaletteEffects/GlobalPaletteEffect.cs

```diff
@@ -0,0 +1,60 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Used for day/night effects.")]
+	class GlobalPaletteEffectInfo : ITraitInfo
+	{
+		public readonly string[] ExcludePalettes = { "cursor", "chrome", "colorpicker", "fog", "shroud" };
+
+		public readonly int Red = 30;
+		public readonly int Green = 30;
+		public readonly int Blue = 90;
+
+		public readonly float Lerp = 0.8f;
+
+		public object Create(ActorInitializer init) { return new GlobalPaletteEffect(this); }
+	}
+
+	class GlobalPaletteEffect : IPaletteModifier
+	{
+		readonly GlobalPaletteEffectInfo info;
+
+		public GlobalPaletteEffect(GlobalPaletteEffectInfo info)
+		{
+			this.info = info;
+		}
+
+		public void AdjustPalette(IReadOnlyDictionary<string, MutablePalette> palettes)
+		{
+			foreach (var kvp in palettes)
+			{
+				if (info.ExcludePalettes.Contains(kvp.Key))
+					continue;
+
+				var palette = kvp.Value;
+
+				for (var x = 0; x < 256; x++)
+				{
+					var from = palette.GetColor(x);
+					var lum = from.GetBrightness();
+					var to = Color.FromArgb(from.A, (int)(lum * info.Red), (int)(lum * info.Green), (int)(lum * info.Blue));
+					palette.SetColor(x, Exts.ColorLerp(info.Lerp, from, to));
```

Comment:
For compatibility with TS, please use:
```csharp
var red = (int)(from.R * info.Ambient * info.Red).Clamp(0, 255);
var green = (int)(from.G * info.Ambient * info.Green).Clamp(0, 255);
var blue = (int)(from.B * info.Ambient * info.Blue).Clamp(0, 255);
palette.SetColor(x, Color.FromArgb(from.A, red, green, blue));
```

and then define in the Info:
```csharp
public readonly float Red = 1f;
public readonly float Green = 1f;
public readonly float Blue = 1f;
public readonly float Ambient = 1f;
```

and add "alpha" to the list of excluded palettes.